apps/rush-lib/src/logic/operations/OperationExecutionManager.ts

```diff
@@ -548,4 +588,131 @@ export class OperationExecutionManager {
     this._terminal.writeStdoutLine(leftPart + rightPart);
     this._terminal.writeStdoutLine('');
   }
+
+  /**
+   * Print a more detailed timeline and analysis of CPU usage for the build.
+   */
+  private _printTimeline(): void {
+    //
+    // Gather the operation records we'll be displaying
+    //
+
+    const operations: OperationExecutionRecord[] = [];
+
+    for (const operation of this._executionRecords) {
+      if (operation.stopwatch.startTime && operation.stopwatch.endTime) {
+        operations.push(operation);
+      }
+    }
+
+    operations.sort((a, b) => a.stopwatch.startTime! - b.stopwatch.startTime!);
+
+    //
+    // Determine timing for all tasks (wall clock and execution times)
+    //
+
+    const allStart: number = operations[0].stopwatch.startTime!;
+    const allEnd: number = Math.max(...operations.map((operation) => operation.stopwatch.endTime!));
+    const allDuration: number = allEnd - allStart;
+    const workDuration: number = operations
+      .map((operation) => operation.stopwatch.duration)
+      .reduce((sum, value) => sum + value);
+
+    //
+    // Do some calculations to determine what size timeline chart we need.
+    //
+
+    const maxNameLength: number = Math.max(...operations.map((operation) => operation.name.length));
+    const maxDurationLength: number = Math.max(
+      ...operations.map((operation) => operation.stopwatch.duration.toFixed(1).length + 1)
+    );
+    const chartWidth: number = TIMELINE_WIDTH - maxNameLength - maxDurationLength - 3;
+    //
+    // Loop through all operations, assembling some statistics about operations and
+    // phases, if applicable.
+    //
+
+    const output: string[] = [];
```

Comment:
What's the point of writing the lines to an array instead of piping them directly to the terminal?

Reply:
In hindsight, the extra string array really isn't giving me anything -- I'll convert it to write each line as it goes.